### PR TITLE
Remove Data Binding

### PR DESCRIPTION
**Pull request by zsmb13**
### Description

Removes Data Binding from both the library and the sample app, replacing its usages with ViewBinding. Also removes the unused `UserActivity` from the library.

Bonus: cleans up some markdown in the README.

### Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] PR targets the `develop` branch
- [x] ~Changelog updated with client-facing changes~
- [x] ~New code is covered by unit tests~
- [x] ~Comparison screenshots added for visual changes~
- [x] Reviewers added

### CODE REVIEW

File: library/src/main/java/com/getstream/sdk/chat/view/activity/UsersActivity.java [hunk 1]
```diff
@@ -1,311 +0,0 @@
...
-import io.getstream.chat.android.client.models.User;
-import io.getstream.chat.android.client.utils.FilterObject;
-import io.getstream.chat.android.client.utils.Result;
-import kotlin.Unit;
-import kotlin.jvm.functions.Function1;
-
-/**
- * An Activity of user list.
- */
-public class UsersActivity extends AppCompatActivity {
```
Comment (ogkuzmin) [inline library/src/main/java/com/getstream/sdk/chat/view/activity/UsersActivity.java:44]: I haven't noticed that we have something like that :-)

**Review by ogkuzmin — APPROVED** [review]
Good changes. Let's get rid of it!)

**Comment by tschellenbach** [comment]
good, databinding is very poorly implemented on android. nice to get rid of it.

**Review by adasiewiczr — APPROVED** [review]
LGTM

**Pull request merged**